pod.go

```diff
@@ -152,6 +155,18 @@ func (pc *podCollector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- descPodContainerResourceLimitsMemoryBytes
 }
 
+func printControllers(annotation map[string]string) string {
+	value, ok := annotation[api.CreatedByAnnotation]
+	if ok {
+		var r api.SerializedReference
+		err := json.Unmarshal([]byte(value), &r)
```

Comment:
Maybe the Kubernetes Decoder would be better: "k8s.io/apimachinery/pkg/util/yaml" with function "NewYAMLOrJSONDecoder" as this JSON was marshaled by the Kubernetes encoder.